### PR TITLE
Add a shared Scheme for EarlGrey.xcodeproj and modify docs to ask not to run the setup script manually.

### PR DESCRIPTION
**Pull request by tirodkar**
Required for https://github.com/google/EarlGrey/issues/30.

### CODE REVIEW

**Comment by bootstraponline** [comment]
@tirodkar this still doesn't work due to https://github.com/google/EarlGrey/issues/58 Carthage need xcodebuild to succeed.

```
$ xcodebuild -project EarlGrey.xcodeproj -configuration Release -target EarlGrey
=== BUILD TARGET EarlGrey OF PROJECT EarlGrey WITH CONFIGURATION Release ===

Check dependencies
CodeSign error: code signing is required for product type 'Framework' in SDK 'iOS 9.3'

** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
```

**Comment by bootstraponline** [comment]
```
$ carthage bootstrap
*** No Cartfile.resolved found, updating dependencies
*** Fetching EarlGrey
*** Checking out EarlGrey at "8ef8f3aaf06f24b5021382a0d8bd01fa9373d995"
*** xcodebuild output can be found in /var/folders/js/59rv7l2j35z5jptfcp0t3vsw0000gn/T/carthage-xcodebuild.oJJ29q.log
*** Building scheme "EarlGrey" in EarlGrey.xcodeproj
** BUILD FAILED **


The following build commands failed:
	CompileC /Users/bootstraponline/Library/Developer/Xcode/DerivedData/EarlGrey-gqzrpuodmexnqzaqgqqsrvccsgvy/Build/Intermediates/EarlGrey.build/Release-iphoneos/EarlGrey.build/Objects-normal/armv7/GREYKeyboard.o EarlGrey/Core/GREYKeyboard.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
/Users/bootstraponline/Desktop/code/carthage_check/Carthage/Checkouts/EarlGrey/EarlGrey.pch:27:4: error: "This project requires NS_BLOCK_ASSERTIONS to be undefined."
A shell task failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	CompileC /Users/bootstraponline/Library/Developer/Xcode/DerivedData/EarlGrey-gqzrpuodmexnqzaqgqqsrvccsgvy/Build/Intermediates/EarlGrey.build/Release-iphoneos/EarlGrey.build/Objects-normal/armv7/GREYKeyboard.o EarlGrey/Core/GREYKeyboard.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
```

```
$ cat Cartfile.private 
github "tirodkar/EarlGrey" "carthage-scheme"
```

**Comment by bootstraponline** [comment]
> /usr/bin/xcrun xcodebuild -project /Users/bootstraponline/Desktop/code/carthage_check/Carthage/Checkouts/EarlGrey/EarlGrey.xcodeproj -scheme EarlGrey -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build

^ that's how carthage is building EarlGrey (fails on NS_BLOCK_ASSERTIONS error).

**Comment by tirodkar** [comment]
@bootstraponline, we've changed Code Signing to "iPhone Distribution" from "Don't Code Sign". Thanks for the command. I'm running it and checking for issues.

**Comment by bootstraponline** [comment]
> @bootstraponline, we've changed Code Signing to "iPhone Distribution" from "Don't Code Sign".

Please use `"iPhone Developer"` as is done in the [WebDriverAgent project](https://github.com/facebook/WebDriverAgent/blob/a19fb8fd0e854d0c947262c4da9f363acaa29fb0/WebDriverAgent.xcodeproj/project.pbxproj#L2062). I'm able to run xcode build with `"iPhone Developer"`.

` "iPhone Distribution"` errors with:
> Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) were found.

**Comment by tirodkar** [comment]
@bootstraponline, on testing it with `iPhone Developer`, the `NS_BLOCK_ASSERTIONS` issue wasn't fixed. Could you confirm?

**Comment by bootstraponline** [comment]
> @bootstraponline, on testing it with iPhone Developer, the NS_BLOCK_ASSERTIONS issue wasn't fixed. Could you confirm?

Yes, I confirm it doesn't fix the assertions error. However the **Code Sign error** is fixed.

```
In file included from <built-in>:359:
In file included from <command line>:3:
/Users/bootstraponline/Desktop/code/carthage_check/Carthage/Checkouts/EarlGrey/EarlGrey.pch:27:4: error: "This project
      requires NS_BLOCK_ASSERTIONS to be undefined."
  #error "This project requires NS_BLOCK_ASSERTIONS to be undefined."
   ^
1 error generated.

** BUILD FAILED **
```

**Comment by bootstraponline** [comment]
`xcodebuild -project EarlGrey.xcodeproj -configuration Release -target EarlGrey` has the same `NS_BLOCK_ASSERTIONS` error.

**Comment by bootstraponline** [comment]
I don't know if this is correct but I have fixed the issue like this:

xcode_assert_fix

**Comment by tirodkar** [comment]
@bootstraponline, adding the flags worked on my side as well. Is there any way we could pass those in the xcodebuild command rather than modify the project itself? The Code Signing Flags have been updated.

**Comment by bootstraponline** [comment]
> Is there any way we could pass those in the xcodebuild command rather than modify the project itself?

Not that I know of. The issue is Carthage uses the project settings, we don't have any direct control over the xcodebuild command.

**Comment by tirodkar** [comment]
@bootstraponline, updated the PR. Take a look if this works fine.

**Comment by bootstraponline** [comment]
It works!!

```
mac:carthage_check bootstraponline$ carthage checkout
*** Fetching EarlGrey
*** Checking out EarlGrey at "036c53169c9e8f0081de0e6ee50baef135138eba"
mac:carthage_check bootstraponline$ carthage build
*** xcodebuild output can be found in /var/folders/js/59rv7l2j35z5jptfcp0t3vsw0000gn/T/carthage-xcodebuild.IuLm3C.log
*** Building scheme "EarlGrey" in EarlGrey.xcodeproj
*** Building scheme "OCMock" in OCMock.xcodeproj
*** Building scheme "OCMock tvOS" in OCMock.xcodeproj
*** Building scheme "OCMock iOS" in OCMock.xcodeproj
mac:carthage_check bootstraponline$ 
```

File: EarlGrey.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1018,6 +1038,8 @@
 				OTHER_CFLAGS = (
 					"-fobjc-arc",
 					"-fobjc-arc-exceptions",
+					"-DNS_BLOCK_ASSERTIONS=1",
```
Comment (khandpur) [inline EarlGrey.xcodeproj/project.pbxproj]: is -DXXX necessary?

You can also build via:
`xcodebuild test OTHER_CFLAGS=XXX
`
Although I am not sure if it appends or replaces whatever is read from the project. You want the former.

Ref: http://clang.llvm.org/docs/CommandGuide/clang.html

Comment (tirodkar) [inline EarlGrey.xcodeproj/project.pbxproj]: As @bootstraponline stated, the issue seems to be that Carthage does not provide direct control over the build flags. @bootstraponline, is there a way by which we could add these parameters in any other way, using a custom script or the scheme or so.

Comment (bootstraponline) [inline EarlGrey.xcodeproj/project.pbxproj]: > is -DXXX necessary?

I checked and you're correct, the build works without it.

> @bootstraponline, is there a way by which we could add these parameters in any other way, using a custom script or the scheme or so.

There might be, but not that I know of.

Comment (bootstraponline) [inline EarlGrey.xcodeproj/project.pbxproj]: Maybe: `xcodebuild -scheme "EarlGrey" -showBuildSettings > EarlGrey.xcconfig`?

Comment (derrh) [inline EarlGrey.xcodeproj/project.pbxproj]: If the framework does require `NS_BLOCK_ASSERTIONS` to be undefined, is there any reason why build settings couldn't explicitly undefine it? I guess I'm just wondering what the hesitation is with adding the flag.

File: EarlGrey.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -973,6 +991,8 @@
...
```
Comment (derrh) [inline EarlGrey.xcodeproj/project.pbxproj]: This line is not required. No reason to define it, just to turn around and undefine it.

**Pull request merged**